crawlers/lib/platforms/gitea.py

```diff
@@ -40,9 +43,13 @@ def crawl(self, state: dict = None) -> Tuple[bool, List[dict], dict]:
                                    f"- response not ok, status: {response.status_code}")
                     return False, [], state  # nr.1 - we skip rest of this block, hope we get it next time
                 result = response.json()
+            except (MaxRetryError, ConnectionError, Timeout, TooManyRedirects) as e:
+                logger.exception(f"{self} - crawler cannot reach hoster")
+                # we re-raise these, as we want to avoid returning empty results to the indexer
+                raise e
             except Exception as e:
                 logger.exception(f"(skipping block chunk) gitea crawler crashed")
-                return False, [], state  # nr.2 - we skip rest of this block, hope we get it next time
+                return False, [], state, e  # nr.2 - we skip rest of this block, hope we get it next time
```

Comment:
First except with specific exceptions are the ones we want to re-raise as they mean we have a complete failure for the block - we could try to salvage these requests more but I think we should start like this.

the other side, all other exceptions are like I said handled like before, we catch them and continue (but now we also yield any caught exceptions that occured).